src/core/manager.c

```diff
@@ -3264,6 +3267,56 @@ void manager_override_watchdog(Manager *m, WatchdogType t, usec_t timeout) {
         m->watchdog_overridden[t] = timeout;
 }
 
+int manager_set_watchdog_pretimeout_governor(Manager *m, const char *governor) {
+        _cleanup_free_ char *p = NULL;
+        int r;
+
+        assert(m);
+
+        if (MANAGER_IS_USER(m))
+                return 0;
```

Comment:
is this the right approach?

right now you just allow any governor to be set in usermode, but don't actually do it. I think it either should return an error, or actually be written to the Manager's `watchdog_pretimeout_governor` and then returned back if requested, though not actually do anything.

Reply:
This mirrors how all other watchdog settings are handled - I don't think this should be treated any differently? This one returns an error mainly because it could go OOM, while the others are not doing memory allocations